### PR TITLE
[converter] implement aten::atan2 conversion

### PR DESCRIPTION
**Pull request by gokulkrishna98**
## Description

Adds a converter for aten.atan2.default. coreai has no native atan2 op, so the conversion decomposes it into atan(y/x) with per-quadrant correction:

- x > 0: atan(y/x)
- x < 0, y ≥ 0: atan(y/x) + π
- x < 0, y < 0: atan(y/x) − π
- x = 0, y > 0: π/2
- x = 0, y < 0: −π/2
- x = 0, y = 0: 0

Division by zero when x=0 is guarded with broadcasting_where before the divide, then the x=0 result is selected in a final where at the end.

## Testing:
- Adds numerical tests (shapes 1D/2D/3D, float32/float16, static/dynamic, axis-aligned edge cases) and IR FileCheck tests (static, dynamic, 1D).
- python unit tests.

### CODE REVIEW

File: coreai_torch/_aten_to_core.py [hunk 1]
```diff
@@ -1547,6 +1547,111 @@ def replace_argmax(values_map: dict[str, Value], node: fx.Node, loc: Location) -
...
+    half_pi = coreai.constant(np.pi / 2.0, dtype=ele_type)
+    neg_half_pi = coreai.constant(-np.pi / 2.0, dtype=ele_type)
+    quarter_pi = coreai.constant(np.pi / 4.0, dtype=ele_type)
+    neg_quarter_pi = coreai.constant(-np.pi / 4.0, dtype=ele_type)
+    three_quarter_pi = coreai.constant(3.0 * np.pi / 4.0, dtype=ele_type)
+    neg_three_quarter_pi = coreai.constant(-3.0 * np.pi / 4.0, dtype=ele_type)
+
+    # ── signed-zero-aware sign predicates ─────────────────────────────────────
+    # 1 / -0.0 = -inf (IEEE-754), so (0 > 1/v) is True iff v = -0.0. Combine with
+    # the strict > predicate (handles ±inf and non-zero finites) via OR.
```
Comment (TobyRoseman) [inline coreai_torch/_aten_to_core.py:1584]: What's OR?

Comment (gokulkrishna98) [reply coreai_torch/_aten_to_core.py:1584]: Hi Toby, it is the `coreai.broadcasting_or` operation used in the computation of y_neg.

y_neg = "y is negative in the usual sense" OR "y is the special IEEE value negative-zero, which > comparisons can't distinguish from positive zero."

**Review by jakesabathia2 — APPROVED** [review]
